File: educational-system-consumer-teacher-9002/src/main/java/edu/hubu/consumerteacher/service/ITeacherService.java
package edu.hubu.consumerteacher.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.teacher.Teacher;
import io.swagger.annotations.ApiImplicitParam;
import io.swagger.annotations.ApiImplicitParams;
import io.swagger.annotations.ApiOperation;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "teacher-provider-service")
public interface ITeacherService {

    @GetMapping("/teacher/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/teacher/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "teacherId") String orderBy);

    @GetMapping("/teacher/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "teacherId") String orderBy
    );

    @PostMapping("/teacher/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "teacherId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.Teacher teacher
    );

    @PostMapping("/teacher/")
    CommonResult save(@RequestBody Teacher teacher);

    @PutMapping("/teacher/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody Teacher teacher);

    @DeleteMapping("/teacher/{id}")
    CommonResult logicDeleteById(@PathVariable("id") String id);

    @PatchMapping("/teacher/{id}")
    CommonResult forbidById(@PathVariable("id") String id);

    @DeleteMapping("/teacher/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/teacher/ITeacherService.kt
package edu.hubu.commons.service.teacher

import edu.hubu.commons.model.teacher.Teacher;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface ITeacherService {
    fun save(t: Teacher): Teacher
    fun logicalDeleteById(id: String): Boolean
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: Teacher): Teacher
    fun forbidById(id: String): Boolean
    fun selectOneById(id: String): Teacher
    fun selectAllOrderBy(order: String = "teacherId"): List<Teacher>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "teacherId"): List<Teacher>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "teacherId", t: edu.hubu.commons.mybatis.model.Teacher): List<edu.hubu.commons.mybatis.model.Teacher>
}
